src/modules/sdk/objections/service.ts

```diff
@@ -0,0 +1,82 @@
+import { AxiosError, AxiosRequestConfig, AxiosResponse, Method } from "axios";
+import axios from "axios";
+import logger from "../../../utils/logger";
+import { INTERNAL_API_URL } from "../../../utils/properties";
+import { ApiError, ObjectionPatch } from "./types";
+
+const HTTP_POST: Method = "post";
+
+/**
+ * An axios config with common elements for API calls.
+ * @param {Method} httpMethod the http method to use eg. get, post
+ * @param {string} url of the api endpoint
+ * @param {string} token Bearer token for API call
+ */
+const getAxiosRequestConfig = (httpMethod: Method, url: string, bearerToken: string): AxiosRequestConfig => {
+  return {
+    headers: {
+      Accept: "application/json",
+      Authorization: "Bearer " + bearerToken,
+    },
+    method: httpMethod,
+    proxy: false,
+    url,
+  };
+};
+
+/**
+ * Call the API using the supplied Axios request config.
+ * @param {AxiosRequestConfig} axios request config
+ * @returns {Promise<AxiosResponse>} the api response
+ * @throws {ApiError}
+ */
+const makeAPICall = async (config: AxiosRequestConfig): Promise<AxiosResponse> => {
+  try {
+    logger.debug(`Calling ${config.method} ${config.url} with data ${config.data}`);
+    const axiosResponse: AxiosResponse = await axios.request<any>(config);
+    logger.debug(`data returned from axios api call : ${JSON.stringify(axiosResponse.data, null, 2)}`);
+    return axiosResponse;
+  } catch (err) {
+    logger.error(`ERROR calling API ${JSON.stringify(err, null, 2)}`);
+    const axiosError = err as AxiosError;
+    const { response, message } = axiosError;
+    throw {
+      data: response ? response.data.errors : [],
+      message,
+      status: response ? response.status : -1,
```

Comment:
The -1 is 'magic'! Could you define a constant explaining what this represents? (probably just a general error?). Maybe the definition should live in the types.ts file?

Reply:
i'll look into that

Reply:
I've changed it to a constant STATUS_NO_RESPONSE now and exported it. Didn't think it warranted being a 'type'